### PR TITLE
Add regression tests for #2441

### LINKED ISSUE #2441: EE uses moduleSymbol.GetReferencedAssemblies() in GetIndexOfReferencedAssembly
The compiler used to do this, but it's inappropriate since it preemptively unifies assemblies, meaning GetIndexOfReferencedAssembly may not find the assembly in the list. The EE should check if this is also a change that should be made there.

See https://github.com/dotnet/roslyn/commit/4d9f92693b7502edd42cf87d17fe8fe502e2ed37 for more information.

### PR DESCRIPTION
**Pull request by amcasey**
After discussing the matter, @agocke, @tmat and I have concluded that the
implementation in EETypeNameDecoder will never be affected by the bug the
compiler was seeing.  However, we still felt it would be worthwhile to
make the implementations consistent.  There is, however, no test, since
the behavior is unchanged.

I added some using/import tests since they use the compiler implementation
(which has already been fixed).

Fixes #2441

### CODE REVIEW

**Comment by amcasey** [comment]
FYI @tmat, @agocke, @cston, @KevinH-MS, @ManishJayaswal

File: src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/EETypeNameDecoder.cs [hunk 1]
```diff
@@ -20,9 +20,8 @@ internal sealed class EETypeNameDecoder : TypeNameDecoder<PEModuleSymbol, TypeSy
 
         protected override int GetIndexOfReferencedAssembly(AssemblyIdentity identity)
         {
-            var assemblies = this.GetAssemblies();
             // Find assembly matching identity.
-            int index = assemblies.IndexOf((assembly, id) => id.Equals(assembly.Identity), identity);
+            int index = this.Module.GetReferencedAssemblies().IndexOf(identity);
             if (index >= 0)
```
Comment (cston) [inline src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/EETypeNameDecoder.cs]: Is this change necessary?

Comment (cston) [inline src/ExpressionEvaluator/CSharp/Source/ExpressionCompiler/EETypeNameDecoder.cs]: Is this change necessary? It assumes that GetReferencedAssemblies and GetReferencedAssemblySymbols are aligned, which the previous code did not.
